doc/docs/master/concepts/pipeline-concepts/pipeline/spout.md

```diff
@@ -83,3 +83,27 @@ The following text is an example of a minimum specification:
   }
 }
 ```
+
+## Resuming Spout Progress
+
+When a spout container crashes, all incomplete operations
+that were processed before the crash are lost, and the spout needs
+to start the interrupted data operation from scratch.
+To keep the history of changes, so that the spout can
+continue where it left off after the restart, you can
+configure a record tracking `marker` file for your spout.
+
+When you specify the `marker` parameter in the
+`transform` section of your pipeline, Pachyderm creates
+the `marker` file or directory. The current spout marker
+is stored in `pfs/out/marker`. The previous marker is stored
+in `pfs/marker`. If a spout container crashes and then starts
+again, it can read the `marker` file and resume where it left
+off instead of starting over.
+
+Markers are useful if you want to leverage a record tracking
+functionality of an external messaging system, such as
+Apache@ Kafka offset management or similar.
```

Comment:
is the @ supposed to be a trademark symbol or something?

Reply:
yes, it should be ® =), my bad. Good catch!